lib/lf/uscalfe/loc_comp_ellbvp.h

```diff
@@ -346,8 +368,37 @@ class MassEdgeMatrixProvider {
         edge_sel_(std::move(edge_selector)),
         fe_precomp_() {
     auto fe = fe_space->ShapeFunctionLayout(base::RefEl::kSegment());
+    LF_ASSERT_MSG(fe != nullptr, "No shape functions specified for edges");
```

Comment:
I think this assert is very important, I don't know why I haven't added this before. Sorry.